sdk/include/opentelemetry/sdk/trace/exporter.h

```diff
@@ -53,3 +53,27 @@ class SpanExporter
 }  // namespace trace
 }  // namespace sdk
 OPENTELEMETRY_END_NAMESPACE
+
+#if defined(OPENTELEMETRY_EXPORT)
+
+namespace std
+{
+
+//
+// Partial specialization of default_delete used by unique_ptr or shared_ptr.
+// This makes the delete of the type in unique_ptr/shared_ptr happening in the
+// DLL in which it is allocated.
+//
+template <>
```

Comment:
should we have this definition only specifically for DLL scenario ?

Reply:
In line 57, there is a check on macro `OPENTELEMETRY_EXPORT`, so this is only defined for DLL scenario. Or anything I missed?